src/app/pages/products-list/card/product-card.component.html

```diff
@@ -0,0 +1,20 @@
+<mat-card *ngIf="product" class="product-card">
+    <mat-card-header>
+        <mat-card-title>{{ product.name }}</mat-card-title>
+    </mat-card-header>
+    <img
+        *ngIf="product.images?.length"
+        mat-card-image
+        [src]="product.images[0].url"
+        [alt]="'Photo of product'"
+    />
```

Comment:
А вот мы такого еще не знаем)
Конечно шутка, ок, что знаешь и заиспользовал `ngIf`, хотя интересней было бы и без него. Однако использование тогда не совсем "докручено" - зачем нам проверять длину массива? Мы же можем проверить наличие первого элемента в индексе и использовать его если он есть:
```suggestion
    <img
        *ngIf="product.images?.[0] as firstImage"
        mat-card-image
        [src]="firstImage.url"
        [alt]="'Photo of product'"
    />
```

А вообще, как по мне, структурная директива тут ни к чему, она лишь усложняет логику шаблона. Я бы поставил тут optional chaining, тогда бы мы не потеряли статичность элемента в шаблоне, что влечет за собой упрощение рендеринга и логики
```suggestion
    <img
        mat-card-image
        [src]="product.images?.[0]?.url"
        [alt]="'Photo of product'"
    />
```

Reply:
Да, так лучше. Только у меня подсвечивается предупреждение под каждым '.?':
`The left side of this optional chain operation does not include 'null' or 'undefined' in its type, therefore the '?.' operator can be safely removed`

Предлагается вообще убрать '.?', хотя очень странно почему...

Reply:
Строгости не хватает этому TS) Если проставить нужный конфиг, то обращение к `product.images[0]` будет отдавать `IProductImage | undefined` и тогда optional chaining будет актуален.
В нашем случае этот конфиг не включен и `product.images[0]` возвращает `IProductImage` и мы optional chaining не оправдывает тут свое использование, т.к. TS видит, что там нет ни `null` ни `undefined`

Reply:
В общем, ты совершенно прав, по хорошему тут должен быть optional chaining, но для этого кейса у нас TS не настроен так строго)